src/controller/main.h

```diff
@@ -12,15 +12,37 @@
 
 //#define DEBUG_UART
 
-#define PWM_CYCLES_COUNTER_MAX                    3125    // 5 erps minimum speed; 1/5 = 200ms; 200ms/64us = 3125
-#define PWM_CYCLES_SECOND                         15625L  // 1 / 64us(PWM period)
-#define PWM_DUTY_CYCLE_MAX                        254
-#define PWM_DUTY_CYCLE_MIN                        20
-#define MIDDLE_PWM_DUTY_CYCLE_MAX                 (PWM_DUTY_CYCLE_MAX/2)
 
 
-// motor rotor offset 
-#define MOTOR_ROTOR_OFFSET_ANGLE                  10
+// motor 
+#define PWM_CYCLES_COUNTER_MAX                                    3125    // 5 erps minimum speed; 1/5 = 200ms; 200ms/64us = 3125
+#define PWM_CYCLES_SECOND                                         15625L  // 1 / 64us(PWM period)
+#define PWM_DUTY_CYCLE_MAX                                        254
+#define PWM_DUTY_CYCLE_MIN                                        0
+#define MIDDLE_PWM_DUTY_CYCLE_MAX                                 127     // PWM_DUTY_CYCLE_MAX / 2
+
+#define PWM_DUTY_CYCLE_RAMP_UP_INVERSE_STEP                       60      // 60 -> 60 * 64 us for every duty cycle increment
+#define PWM_DUTY_CYCLE_RAMP_DOWN_INVERSE_STEP                     25      // 25 -> 25 * 64 us for every duty cycle increment
+
+/*---------------------------------------------------------
+  NOTE: regarding duty cycle (PWM) ramping
+  
+  Choose appropriate duty cycle (PWM) ramp up/down step. 
+  A higher value will make the motor acceleration slower.
+  
+  For a 24 V battery, 25 for ramp up seems ok. For a 
+  higher voltage battery, this values should be higher.
+---------------------------------------------------------*/
+
+
+
+#define MOTOR_ROTOR_OFFSET_ANGLE                                  10
+#define MOTOR_ROTOR_ANGLE_90                                      73      // (63  + MOTOR_ROTOR_OFFSET_ANGLE)
```

Comment:
Just to clarify again, does SDCC really not optimize things like (63 + MOTOR_ROTOR_OFFSET_ANGLE) ?

Reply:
Even if it does optimize things like that, it does not hurt to change it. I trust Casainho when he says that differences such as these make a difference.

Reply:
If it actually makes a difference in the generated code by SDCC, by all means we should change these constants.
The downside of defining constants that depend on other constants like this, is that they get out of sync. e.g. if someone changes MOTOR_ROTOR_OFFSET_ANGLE, how will you ensure they don't forget to change all the other related defines ?
In this case they're right next to each other, but what if there are other defines depending on this in other header files ?
That's why, if SDCC does actually do proper constant propagation, it's probably better to leave the calculations in.

(I'll check later to see when and where SDCC does or doesn't do constant propagation to actually get this discussion settled ;) )

Reply:
As they are close to each other it is okay in this case. But this is NOT something one should change if not knowing what they are changing. They should know to change the other values as well. 

Please do check and settle this. :)

Reply:
Just checked on 0.19.0, compiled once with the old style (MOTOR_ROTOR_OFFSET + something), and once with constants pre-calculated. Both result in the exact same hexfile.

Full output:
```
[frans-willem@Hydra controller]$ sdcc --version
SDCC : mcs51/z80/z180/r2k/r3ka/gbz80/tlcs90/ez80_z80/ds390/pic16/pic14/TININative/ds400/hc08/s08/stm8/pdk14/pdk15 3.9.0 #11195 (Linux)
published under GNU General Public License (GPL)
[frans-willem@Hydra controller]$ cat main.h | grep ROTOR_ANGLE
#define MOTOR_ROTOR_ANGLE_90                      73
#define MOTOR_ROTOR_ANGLE_150                     116
#define MOTOR_ROTOR_ANGLE_210                     158
#define MOTOR_ROTOR_ANGLE_270                     201
#define MOTOR_ROTOR_ANGLE_330                     243
#define MOTOR_ROTOR_ANGLE_30                      30
[frans-willem@Hydra controller]$ rm -rf main.hex
[frans-willem@Hydra controller]$ make -f Makefile_linux >/dev/null 2>&1
[frans-willem@Hydra controller]$ cp main.hex main_constants.hex
[frans-willem@Hydra controller]$ git checkout main.h
Updated 1 path from the index
[frans-willem@Hydra controller]$ cat main.h | grep ROTOR_ANGLE
#define MOTOR_ROTOR_ANGLE_90                      (63  + MOTOR_ROTOR_OFFSET_ANGLE)
#define MOTOR_ROTOR_ANGLE_150                     (106 + MOTOR_ROTOR_OFFSET_ANGLE)
#define MOTOR_ROTOR_ANGLE_210                     (148 + MOTOR_ROTOR_OFFSET_ANGLE)
#define MOTOR_ROTOR_ANGLE_270                     (191 + MOTOR_ROTOR_OFFSET_ANGLE)
#define MOTOR_ROTOR_ANGLE_330                     (233 + MOTOR_ROTOR_OFFSET_ANGLE)
#define MOTOR_ROTOR_ANGLE_30                      (20  + MOTOR_ROTOR_OFFSET_ANGLE)
[frans-willem@Hydra controller]$ rm -rf main.hex
[frans-willem@Hydra controller]$ make -f Makefile_linux >/dev/null 2>&1
[frans-willem@Hydra controller]$ cp main.hex main_expressions.hex
[frans-willem@Hydra controller]$ diff -s main_constants.hex main_expressions.hex
Files main_constants.hex and main_expressions.hex are identical
```

Reply:
That is great actually and thank you for taking the time to test it out! Will fix in the future!